src/test/groovy/com/bmuschko/gradle/docker/tasks/image/DockerfileTest.groovy

```diff
@@ -7,47 +7,66 @@ import static com.bmuschko.gradle.docker.tasks.image.Dockerfile.*
-        new OnBuildInstruction({ 'ENV JAVA_HOME /usr/java' })                           | 'ONBUILD'    | 'ONBUILD ENV JAVA_HOME /usr/java'
-        new LabelInstruction(['description': 'Single label' ])                          | 'LABEL'      | 'LABEL "description"="Single label"'
-        new LabelInstruction({ ['description': 'Single label' ] })                      | 'LABEL'      | 'LABEL "description"="Single label"'
-        new LabelInstruction(['description': 'Multiple labels', 'version': '1.0' ])     | 'LABEL'      | 'LABEL "description"="Multiple labels" "version"="1.0"'
-        new LabelInstruction({ ['description': 'Multiple labels', 'version': '1.0' ] }) | 'LABEL'      | 'LABEL "description"="Multiple labels" "version"="1.0"'
+        instructionInstance                                                             | expectedKeyword | expectedBuiltInstruction
+        new GenericInstruction('FROM ubuntu:14.04')                                     | 'FROM'          | 'FROM ubuntu:14.04'
+        new GenericInstruction({ 'FROM ubuntu:14.04' })                                 | 'FROM'          | 'FROM ubuntu:14.04'
+        new FromInstruction('ubuntu:14.04')                                             | 'FROM'          | 'FROM ubuntu:14.04'
+        new FromInstruction({ 'ubuntu:14.04' })                                         | 'FROM'          | 'FROM ubuntu:14.04'
+        new MaintainerInstruction('John Doe "john.doe@gmail.com"')                      | 'MAINTAINER'    | 'MAINTAINER John Doe "john.doe@gmail.com"'
+        new MaintainerInstruction({ 'John Doe "john.doe@gmail.com"' })                  | 'MAINTAINER'    | 'MAINTAINER John Doe "john.doe@gmail.com"'
+        new RunCommandInstruction('apt-get update && apt-get clean')                    | 'RUN'           | 'RUN apt-get update && apt-get clean'
+        new RunCommandInstruction({ 'apt-get update && apt-get clean' })                | 'RUN'           | 'RUN apt-get update && apt-get clean'
+        new DefaultCommandInstruction('ping google.com')                                | 'CMD'           | 'CMD ["ping google.com"]'
+        new DefaultCommandInstruction({ 'ping google.com' })                            | 'CMD'           | 'CMD ["ping google.com"]'
+        new ExposePortInstruction(8080)                                                 | 'EXPOSE'        | 'EXPOSE 8080'
+        new ExposePortInstruction({ 8080 })                                             | 'EXPOSE'        | 'EXPOSE 8080'
+        new ExposePortInstruction(8080, 9090)                                           | 'EXPOSE'        | 'EXPOSE 8080 9090'
+        new ExposePortInstruction({ [8080, 9090] })                                     | 'EXPOSE'        | 'EXPOSE 8080 9090'
+        new EnvironmentVariableInstruction('OS', 'Linux')                               | 'ENV'           | 'ENV OS Linux'
+        // TODO: negative test:
```

Comment:
you mean the negative test case? Yes, we should test the exceptional case I think ;)